java/code/src/com/suse/manager/reactor/messaging/LibvirtEngineMessage.java

```diff
@@ -0,0 +1,104 @@
+/**
+ * Copyright (c) 2018 SUSE LLC
+ *
+ * This software is licensed to you under the GNU General Public License,
+ * version 2 (GPLv2). There is NO WARRANTY for this software, express or
+ * implied, including the implied warranties of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. You should have received a copy of GPLv2
+ * along with this software; if not, see
+ * http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
+ *
+ * Red Hat trademarks are not licensed under GPLv2. No permission is
+ * granted to use or replicate Red Hat trademarks that are incorporated
+ * in this software or its documentation.
+ */
+package com.suse.manager.reactor.messaging;
+
+import com.redhat.rhn.common.messaging.EventMessage;
+
+import com.google.gson.JsonElement;
+import com.suse.salt.netapi.event.EngineEvent;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+
+/**
+ * Libvirt_events engine message to handle
+ */
+public abstract class LibvirtEngineMessage implements EventMessage {
+
+    private String connection;
+    private Optional<String> minionId;
+    private String timestamp;
+
+    /**
+     * Parse the engine event if it is a libvirt_events one.
+     *
+     * @param engineEvent the event to parse
+     * @return the corresponding event message or <code>null</code> if not handled
+     *
+     * @throws IllegalArgumentException if the event is invalid
+     */
+    public static LibvirtEngineMessage create(EngineEvent engineEvent) throws IllegalArgumentException {
+        List<String> additionalParts = Arrays.asList(engineEvent.getAdditional().split("/"));
+        if (additionalParts.size() < 3) {
```

Comment:
Why 3? What does it represent?
Maybe it's worth to extract it to a variable.

Reply:
it represents the number of path parts in the event tag that are needed for libvirt-events

Reply:
extracted into a constant